subsys/bootloader/cmake/sign.cmake

```diff
@@ -59,9 +59,13 @@ include(${CMAKE_CURRENT_LIST_DIR}/../cmake/bl_validation_magic.cmake)
 
 set(slots s0_image)
 
-if (CONFIG_MCUBOOT_BUILD_S1_VARIANT)
+if (CONFIG_BUILD_S1_VARIANT)
   list(APPEND slots s1_image)
-  set(s1_image_is_from_child_image mcuboot)
+
+  # Only add extra dependency if s1 image is not 'app'.
+  if (NOT CONFIG_S1_VARIANT_IMAGE_NAME STREQUAL "app")
+    set(s1_image_is_from_child_image ${CONFIG_S1_VARIANT_IMAGE_NAME})
```

Comment:
This var isn't used in this file. Please add a comment to say where it's used.

Reply:
It is used, but through a dereferenced variable a few lines below :)

Reply:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1779/files/e50687d27857e794e566aeb35f58845191b940af#diff-3f5f4ea605adc9d9d04932522d239efaR82